File: src/package.rs
pub struct Package {
    pub name: String,
    pub version: String,
    pub installed: bool,
    pub required_deps: Vec<Package>,
    pub recommended_deps: Vec<Package>,
    pub optional_deps: Vec<Package>,
    pub required_run_deps: Vec<Package>,
    pub recommended_run_deps: Vec<Package>,
    pub optional_run_deps: Vec<Package>
}

impl Default for Package {
    fn default() -> Package {
        Package {
            name: String::from(""),
            version: String::from("0.0.0"),
            installed: false,
            required_deps: Vec::from([]),
            recommended_deps: Vec::from([]),
            optional_deps: Vec::from([]),
            required_run_deps: Vec::from([]),
            recommended_run_deps: Vec::from([]),
            optional_run_deps: Vec::from([])
        }
    }
}

impl Package {
    fn to_string(&self) -> String {
        return format!("{}-{}", self.name, self.version);
    }

    fn get_entry(&self) -> String {
        return self.to_string();
    }
}

File: src/package.py
class Package:
    def __init__(self, name, version):
        self.name = name
        self.version = version
        self.installed = False
        self.req_deps = []
        self.rec_deps = []
        self.opt_deps = []
        self.req_run_deps = []
        self.rec_run_deps = []
        self.opt_run_deps = []

    def __to_string__(self):
        return f'{self.name}-{self.version}'

    @property
    def entry(self):
        return self.__to_string__()